clash-lib/src/Clash/Netlist/BlackBox.hs

```diff
@@ -281,10 +282,18 @@ extractPrimWarnOrFail
 extractPrimWarnOrFail nm = do
   prim <- HashMap.lookup nm <$> Lens.use primitives
   case prim of
-    Just guardedPrim ->
-      -- See if we need to warn the user, or error because we encountered
-      -- a primitive the user explicitly requested not to translate
-      go guardedPrim
+    Just (HasBlackBox warnings compiledPrim) ->
+      -- See if we need to warn the user
+      if null warnings then return compiledPrim else go warnings compiledPrim
```

Comment:
This if statement isn't strictly necessary, but this way the same primitives should end up in the `_seenPrimitives` field of the `NetlistState` as was the case before. That might be a little faster.